### ISSUE TITLE
"Auto inventory letters" option maps items to vi keys

### ISSUE THREAD

**Issue by cafeoh** [CleverRaven/Cataclysm-DDA#19340]
As title says, in multiple menus including the inventory screen, the auto assignment algorithm maps items indiscriminately to hjkl.

Assigning those items to another key or unassigning them only works until something else gets picked up.

**Comment by keyspace** [comment]
To clarify: this is a nuisance if `hjkl` are used to navigate in the menus. (Just found out it is possible.)

**Comment by cafeoh** [comment]
Yes, with auto inventory letters enabled, you gotta switch between control schemes all the time. This is particularly annoying for owners of those hybrid TKL keyboards ([my keyboard for reference](http://www.gnd-tech.com/image/i/CXNcE.jpg)) which offers a full numpad at the expense of the arrow keys, and made yet even worse if you don't have a qwerty layout (0-9 key row being assigned to special character for lowercase). I'm undoubtedly an atypical case, but as it stands this is by far my biggest issue with this game's interface.

- Arrow keys are required for movement in the wear/wield/eat/...etc menus.
- vi keys are required for movement in the examine 'e' menu.
- number keys are required for moving partial stacks, and numpad keys are very convenient for movement, especially for driving (I find).

While arrow keys let you navigate almost everything , I find them unpractical (which is why many devs use hjkl in vim and the like), they lack diagonals for character movement, and they conflict in my case with the numpad keys.

"Full" support for the numpad key isn't trivial (we still want to type numbers for partial stack transfer), but I believe the game's interface could be greatly improved by just allowing the use of vi keys everywhere.

Basically when assigning inventory letter, check against local keybindings to avoid item/binding conflicts. For all the people who couldn't care less about vi keys but still want to benefit from some additional easy-to-reach item shortcuts, just unbind them (it doesn't make sense to have them do different thing depending on inventory load anyway).

**Comment by bouchacha** [comment]
I came here to post exactly this issue. The list of items viewed in the environment through 'V' already blacklist 'F' in order to be able to filter. It seems like a simple solution is to _at least_ blacklist any keybindings used for cardinal directions (i.e. HJKL) and prevent those from automatically being assigned to inventory items. You can still assign inventory to whatever letter you want to but it needs to be intentional.

This is particularly annoying when it comes to consuming items. I have often unintentionally eaten the wrong thing because my hand naturally pressed 'J' to scroll down and instead got a mouthful of dry macaroni 😢

**Comment by bdgackle** [comment]
I mocked this up and submitting a pull request.

It adds an option in settings->interface (off by default) that causes j/h/k/l to get skipped when enabled.  This way it represents no change to most users unless they seek it out and enable it.  It seems to work pretty well.

The one possible downside to my implementation is that it doesn't follow key remaps -- it just hard codes the vi keys to be skipped.  Still, I think we're probably in the minority, and people using non-standard vi keys would be a corner case of a corner case.  The advantage of hard coding in this case is that the change from existing code (and thus the chance of introducing new bugs) is very minimal.

**Comment by Night-Pryanik** [comment]
Closing as per lack of discussion.
